src/array/dimension.rs

```diff
@@ -39,6 +36,15 @@ impl<'ctx> Dimension<'ctx> {
         *self.raw
     }
 
+    /// Read from the C API whatever we need to use this dimension from Rust
+    pub(crate) fn load(
```

Comment:
I'm not a fan of using `load` here as our code base generally uses `load` to refer to pulling bytes from disk or network.

I'm pretty sure the more idiomatic approach here would be to either use a From trait (or something like it depending on multiple argument support) or a method prefixed with `from_` if multiple arguments means we go the "static method" approach (no clue what Rust calls those).

Reply:
I think `new` is also commonly used for this purpose :)